.github/workflows/publish-branch.yaml

```diff
@@ -0,0 +1,38 @@
+name: Publish Branch
+
+on:
+    push:
+        branches:
+            - main
+            - dev
+
+jobs:
+    release_candidate:
+        runs-on: ubuntu-latest
+        env:
+            NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+            NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.head_ref }}
+
+            - name: Setup Project
+              uses: ./.github/actions/setup
+
+            - name: Setup npm credentials file
+              run: echo "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN" >> .npmrc
```

Comment:
One small thing, I don't believe we need this step. The NODE_AUTH_TOKEN being set in the environment should be sufficient here.

Reply:
I'll remove it :) 

About the build, on ReactTooltip we have this script on our package.json: `"prepublishOnly": "npm run build"`.

But looks like Pixi.js doesn't have it, do you prefer to add it to the package.json or in the workflow?

Reply:
I'd say add it to the workflow or else we'd need to fix it double-building on the other workflows. We can clean that up later.

Reply:
ok, I'll add it again.

Reply:
Ok, I'll add it into the workflow